vtm/src/org/oscim/utils/geom/TileSeparator.java

```diff
@@ -38,42 +43,56 @@ public void setRect(float xmin, float ymin, float xmax, float ymax) {
     }
 
     /**
-     * Separates a poly geometry from tile (doesn't clip it).
+     * Separates geometry from tile without clipping it. Points and polys only.
      *
      * @param geom the geometry to be separated
      * @return true if geometry is inside the tile, false otherwise
      */
     public boolean separate(GeometryBuffer geom) {
-        if (!geom.isPoly())
-            return false;
 
-        int pointPos = 0;
+        if (geom.isPoint()) {
+            if (geom.index.length > 1 && geom.index[0] == 2) {
+                float cx = geom.points[0];
+                float cy = geom.points[1];
 
-        for (int indexPos = 0, n = geom.index.length; indexPos < n; indexPos++) {
-            int len = geom.index[indexPos];
-            if (len < 0)
-                break;
-
-            if (len < 6) {
-                pointPos += len;
-                continue;
+                if (isInside(cx, cy)) {
+                    return true;
+                }
+            } else {
+                log.warn("Geometry (Point) has wrong format: " + geom.toString());
             }
+        } else if (geom.isPoly()) {
+            int pointPos = 0;
 
-            int end = pointPos + len;
+            for (int indexPos = 0, n = geom.index.length; indexPos < n; indexPos++) {
+                int len = geom.index[indexPos];
+                if (len < 0)
+                    break;
 
-            for (int i = pointPos; i < end; ) {
-                float cx = geom.points[i++];
-                float cy = geom.points[i++];
+                if (len < 6) {
+                    pointPos += len;
+                    continue;
+                }
 
-                if (cx >= xmin && cx < xmax && cy >= ymin && cy < ymax) {
-                    /* current is inside */
-                    return true;
+                int end = pointPos + len;
+
+                for (int i = pointPos; i < end; ) {
+                    float cx = geom.points[i++];
+                    float cy = geom.points[i++];
+
+                    if (isInside(cx, cy)) {
+                        return true;
+                    }
                 }
-            }
 
-            pointPos += len;
+                pointPos += len;
+            }
         }
         geom.clear();
```

Comment:
Why call `clear()` in case of no supported geometry?

Reply:
Geometries (here lines only) which aren't clipped, shouldn't be called anyway. Only wanted to save one `if` statement. But can of course exclude lines from `clear`.

Reply:
Looking the code, when there are not supported geometries OR the supported ones are not inside, i.e. in all false cases, then a clear of geom happens. Question is why?

Reply:
Ah ok, sorry I took it from [TileClipper](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/utils/geom/TileClipper.java#L67), where `geom` is cleared even if it will fail to clip. But that's not exactly the same, as I saw right now... So we can remove it.

Reply:
Depends on the description :grin:
But in our cases it shouldn't make a difference, so `clear` isn't needed.